addons/grenadethrowing/functions/fnc_throw.sqf

```diff
@@ -0,0 +1,171 @@
+/*
+ * Author: Dslyecxi, Jonpas
+ * Throw grenade.
+ *
+ * Arguments:
+ * None
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [] call ace_grenadethrowing_fnc_throw
+ *
+ * Public: No
+ */
+#include "script_component.hpp"
+
+params ["_unit"];
+
+private _currentThrowable = currentThrowable _unit;
+
+// Check to see if we have anything to throw.
+if (count _currentThrowable < 1 || {_currentThrowable select 0 == ""}) exitWith {
+    [_unit, "No valid throwables"] call FUNC(exitThrowMode); // If we've exhausted a type of grenade, currentThrowable select 0 will have a blank string
+};
+
+// Since we have something to throw, let's create it. By activating GrenadeInHand, the PFH creates a grenade
+private _dropType = getText (configFile >> "CfgMagazines" >> _currentThrowable select 0 >> "ammo");
+
+if (_dropType == "") exitWith {
+    [_unit, "No valid throwables (check 2)"] call FUNC(exitThrowMode);
+};
+
+GVAR(ActiveGrenadeType) = _dropType;
+GVAR(CookingGrenade) = false; // Can't be cooking, just pulled it.
+GVAR(CurrentThrowSpeed) = 0;
+GVAR(LastGrenadeTypeChecked) = "";
+GVAR(GrenadeInHand) = true;
+
+// PFH is maintaining at this point, we're just waiting for a throw, or exit if we die.
+[{
+    params ["_unit"];
+    GVAR(ThrowGrenade) || !alive _unit
+}, {
+    params ["_unit"];
+
+    if (GVAR(CancelThrow) || !alive _unit || count (currentThrowable _unit) < 1) exitWith {
+        [_unit, "Throw check (cancel? valid? alive? etc), apparently not!"] call FUNC(exitThrowMode);
+    };
+
+    // Make it real at the end.
+    _unit playAction "ThrowGrenade";
+
+    // If CTRL is held, we don't delay. Otherwise we wait for the playAction to complete, which is roughly 0.3 seconds.
+    private _waitTime = 0.3;
+    if (GVAR(CtrlHeld)) then {
+        _waitTime = 0.01;
+    };
+
+    [{
+        params ["_unit"];
+
+        // If the grenade's not already cooked, create the "real" one
+        if (!GVAR(CookingGrenade)) then {
+            [_unit, GVAR(ActiveGrenadeItem), GVAR(ActiveGrenadeType)] call FUNC(cook);
+        };
+
+        // Handle removing stuff from our inventory, working on BI bugs
+        private _typeCount = 0;
+        private _typeGrenCheck = (currentThrowable _unit) select 0;
+
+        {
+            if (_x == _typeGrenCheck) then {
+                _typeCount = _typeCount + 1;
+            };
+        } forEach (magazines _unit);
+
+        // Works around a bug where removing one will make them unselectable
+        if (_typeCount > 1) then {
+            TRACE_1("Removing Throwable (2)",_typeGrenCheck);
+            _unit removeMagazine _typeGrenCheck;
+            _unit removeMagazine _typeGrenCheck;
+            _unit addMagazine _typeGrenCheck;
```

Comment:
Apparently it was not, just tried removing one with a debug console and all are gone from the vanilla throwing, can bring them back with the already existing cycle grenade keybinds from ACE though.